### PR TITLE
Name NRF flash sync choice

### LINKED ISSUE #28559: Unable to extend the flash sync API part of the BLE Controller
**Describe the bug**
https://github.com/zephyrproject-rtos/zephyr/pull/27459 introduced a choice to select which flash driver sync interface to use.
The intent was to be able extend this out-of-tree. Unfortunately that is not possible because the choice is unnamed.

**To Reproduce**
Try to add a new flash driver sync interface

**Expected behavior**
It is possible to add another sync interface

**Impact**
It is impossible to add new flash driver sync interfaces out-of-tree.

### PR DESCRIPTION
**Pull request by eugmes**
I'm working on alternate NRF flash sync API implementation for `sdk-nrf` and having the API choice unnamed prevents me from adding the alternate implementation.

I have also renamed the current config option so it is not too generic, made it depend on the Zephyr link layer.

Fixes: #28559

### CODE REVIEW

**Review by joerchan — APPROVED** [review]

**Review by rugeGerritsen — APPROVED** [review]

**Review by nvlsianpu — COMMENTED** [review]
Actually I had kept previous name of synchronization option for backward compatibility (if a user has own configuration - nothing has been changing). I was aware that current name is generic to much.

I was supposing that you will just add your `config` to `choice` construction (e.g. `SOC_FLASH_NRF_RADIO_SYNC_NRF_PROPRIETARY`), and will keep `SOC_FLASH_NRF_RADIO_SYNC` as name for vanilla zephyr synchronization.

File: drivers/flash/Kconfig.nrf [hunk 1]
```diff
@@ -20,16 +20,16 @@ menuconfig SOC_FLASH_NRF
 
 if SOC_FLASH_NRF
 
-choice
+choice SOC_FLASH_NRF_RADIO_SYNC_CHOICE
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply drivers/flash/Kconfig.nrf:23]: Named choice is needed in order to be able to extend it
in other modules.

**Review by nvlsianpu — APPROVED** [review]
Commits messages respond good enough to my doubts.
Great job!

**Review by cvinayak — APPROVED** [review]

**Comment by rugeGerritsen** [comment]
@nvlsianpu Can this be put in the 2.4.0 timeline? The configurations were introduced after 2.3.0. It would be nice if we can avoid renaming those between 2.4.0 and 2.5.0.

**Comment by nvlsianpu** [comment]
> @nvlsianpu Can this be put in the 2.4.0 timeline? The configurations were introduced after 2.3.0. It would be nice if we can avoid renaming those between 2.4.0 and 2.5.0.

This would be exception form rule merging only bug-fix while code-freeze.
Although this change is nRF only related and quite of small code area.
@carlescufi Can an exception to the rule be allowed?

**Comment by rugeGerritsen** [comment]
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/28559

**Pull request merged**